src/compiler/checker.ts

```diff
@@ -7082,7 +7082,9 @@ namespace ts {
         function getDefaultConstraintOfConditionalType(type: ConditionalType) {
             if (!type.resolvedDefaultConstraint) {
                 const rootTrueType = type.root.trueType;
-                const rootTrueConstraint = rootTrueType.flags & TypeFlags.Substitution ? (<SubstitutionType>rootTrueType).substitute : rootTrueType;
+                const rootTrueConstraint = rootTrueType.flags & TypeFlags.Substitution
+                    ? ((<SubstitutionType>rootTrueType).substitute).flags & TypeFlags.AnyOrUnknown ? (<SubstitutionType>rootTrueType).typeVariable : getIntersectionType([(<SubstitutionType>rootTrueType).substitute, (<SubstitutionType>rootTrueType).typeVariable])

```

Comment:
what is this for?

Reply:
also, PLEASE make this line a little shorter. I can't read this diff even in VSCode's diff view.
(I also don't like double-nested ternarys of the form ??:: but they are palatable with shorter lines)

Reply:
ok, I remember you telling me about this. Is the new SubtleSkip test intended to test this? I don't see any conditional types with a constraint of any | unknown.

Reply:
Fixing conditional types once the change is in. Without it, a conditional type `T extends any ? T : never` (as in the react `.d.ts`) has a contextual type of `any` instead of `T`.